src/CLU/Microsoft.CLU/CommandLineParser/UnixCommandLineParser.cs

```diff
@@ -183,17 +183,42 @@ public string FormatParameterName(string name, string type, bool isMandatory, bo
             return builder.ToString();
         }
 
-        private void PresentCommandHelp(ICommandBinder commandBinder, string[] arguments, bool prefix)
+        private void PresentCommandCompletion(string[] arguments)
+        {
+            foreach (var cmd in Help.CommandDispatchHelper.CompleteCommands(CLUEnvironment.GetPackagesRootPath(), arguments).OrderBy((cmd) => cmd.Commandline))
+            {
+                Console.WriteLine(cmd.Commandline);
+            }
+        }
+
+        private void PresentCommandHelp(string[] arguments)
         {
             // BUGBUG - NYI!
             if (arguments.Length == 0)
             {
-                var concatenatedArgs = string.Join(" ", arguments);
-                Console.WriteLine($"Look up help corresponding to the arguments '{concatenatedArgs}'");
+                Console.WriteLine("UNDONE: Present help for Azure itself...");
             }
             else
             {
-                Console.WriteLine("Present help for Azure itself...");
+                var helpFile = Help.CommandDispatchHelper.FindBestHelp(CLUEnvironment.GetPackagesRootPath(), arguments);
+                if (helpFile != null)
+                {
+                    // Found appropriate help...
+                    var cmdLine = string.Join(" ", arguments);
+                    Console.WriteLine($"Help for command {cmdLine}");
+                    Console.WriteLine();
+
+                    foreach (var line in helpFile.GetHelpContent())
+                    {
+                        Console.WriteLine(line);
+                    }
+                    return;
+                }
+                else
+                {
+                    var joinedArgs = string.Join(" ", arguments);
+                    Console.WriteLine($"UNDONE - unable to find help for command {joinedArgs}");
```

Comment:
I'm not familiar with UNDONE - is this meant as a TODO?  If so, you should use this format:

TODO <bug-id>: unable to find help for command

At any rate, it doesn't seem like something that should go to the command line